src/pocket/PocketECSCodePipeline.ts

```diff
@@ -0,0 +1,278 @@
+import { Resource } from 'cdktf';
+import { Construct } from 'constructs';
+import {
+  Codepipeline,
+  DataAwsIamPolicyDocument,
+  DataAwsKmsAlias,
+  IamRole,
+  IamRolePolicy,
+  S3Bucket,
+} from '../../.gen/providers/aws';
+
+export interface PocketECSCodePipelineProps {
+  prefix: string;
+  source: {
+    repository: string;
+    branchName: string;
+    codeStarConnectionArn: string;
+  };
+  codeBuildProjectName?: string;
+  codeDeploy?: {
+    applicationName?: string;
+    deploymentGroupName?: string;
+    appSpecPath?: string;
+    taskDefPath?: string;
+  };
+  tags?: { [key: string]: string };
+}
+
+export class PocketECSCodePipeline extends Resource {
+  private static DEFAULT_TASKDEF_PATH = 'taskdef.json';
+  private static DEFAULT_APPSPEC_PATH = 'appspec.json';
+
+  public readonly codePipeline: Codepipeline;
+
+  constructor(
+    scope: Construct,
+    name: string,
+    private config: PocketECSCodePipelineProps
+  ) {
+    super(scope, name);
+
+    this.codePipeline = this.createCodePipeline();
+  }
+
+  /**
+   * Create a CodePipeline that runs CodeBuild and ECS CodeDeploy
+   * @private
+   */
+  private createCodePipeline(): Codepipeline {
+    const pipelineRole = this.createPipelineRole();
+
+    const s3KmsAlias = new DataAwsKmsAlias(this, 'kms_s3_alias', {
+      name: 'alias/aws/s3',
+    });
+
+    const pipelineArtifactBucket = new S3Bucket(this, 'codepipeline-bucket', {
```

Comment:
can you add https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#force_destroy that would allow us to destroy this stack cleanly

Reply:
Should we worry about bucketings existing with this name? Should we add some hash of the name to guarentee uniqueness?

Reply:
Also is the bucket setup to be private?

Reply:
Good point on the name, I will add the hash. Default acl for s3 bucket is `private` but I will make that explicit